src/compiler/emitter.rs

```diff
@@ -209,6 +212,14 @@ impl Printer {
         self.write_punctuation("}");
     }
 
+    fn emit_union_type(&mut self, node: &UnionTypeNode) {
+        self.emit_list(
+            Some(node),
+            Some(&node.types),
+            ListFormat::UnionTypeConstituents,
```

Comment:
This is sort of cool how a bitflags "compound value" is like a set of configuration options, in this case `ListFormat::UnionTypeConstituents` includes `ListFormat::BarDelimited` so when this `ListFormat` gets passed down to `.write_delimiter()` that will see that flag set and write a `|`